ecllibrary/std/Date.ecl

```diff
@@ -1080,6 +1079,86 @@ SHARED WeekNumForDate(Date_t date, Date_t referenceDate, UNSIGNED1 startingDayOf
     RETURN weekNum;
 END;
 
+/**
+ * Returns a number representing the day of the week indicated by the given date in ISO-8601.
+ * The date must be in the Gregorian calendar after the year 1600.
+ *
+ * @param date          A Date_t value.
+ * @return              A number 1-7 representing the day of the week, where 1 = Monday.
+ */
+EXPORT ISODayOfWeekFromDate( Date_t d) := ((DayOfWeek( d ) + 5 ) % 7) + 1;		
+
+/**
+ * Helper function to figure out the number of weeks for a given year.
+ * For more details, see: https://en.wikipedia.org/wiki/ISO_week_date#Weeks_per_year
+ *
+ * @param date          A Date_t value.
+ * @return              The number 52 for short years and 53 for long ones.
+ */
+SHARED ISOWeeksP( UNSIGNED year ) := ( year + (year/4) - (year/100) + (year/400) ) % 7;
+
+/**
+ * Returns true for years with 53 weeks, and false for years with 52 weeks.
+ *
+ * @param date          A Date_t value.
+ * @return              TRUE if year is a long year (i.e. 53 weeks), FALSE otherwise.
+ */
+SHARED ISOIsLongYear( UNSIGNED year ) := (ISOWeeksP( year ) = 4 OR ISOWeeksP( year - 1 ) = 3);
```

Comment:
Well that would allow negative years. I don't know whether that would work in such cases (I'm no ISO-8601 expert but haven't seen anything about it) and that's all I can say.

Reply:
ISO-8601 does not require support for negative years, but it also does not prohibit them.  Early and negative years are problematical in most calendar systems because the calendars themselves changed (e.g. Gregorian vs Julian), and tracking time backwards a year at a time never matches up correctly.  We do support negative years throughout the rest of the module, so we should probably do so here for consistency if not accuracy.  Please go ahead and change that one datatype.

Reply:
Done.